io/io/inc/TBufferFile.h

```diff
@@ -38,74 +38,40 @@ class TVirtualStreamerInfo;
 class TStreamerInfo;
 class TStreamerElement;
 class TClass;
-class TExMap;
 class TVirtualArray;
 namespace TStreamerInfoActions {
    class TActionSequence;
 }
 
-class TBufferFile : public TBuffer {
+class TBufferFile : public TBufferIO {
 
 protected:
    typedef std::vector<TStreamerInfo*> InfoList_t;
 
-   Int_t           fMapCount;      ///< Number of objects or classes in map
-   Int_t           fMapSize;       ///< Default size of map
-   Int_t           fDisplacement;  ///< Value to be added to the map offsets
-   UShort_t        fPidOffset;     ///< Offset to be added to the pid index in this key/buffer.
-   TExMap         *fMap;           ///< Map containing object,offset pairs for reading/writing
-   TExMap         *fClassMap;      ///< Map containing object,class pairs for reading
-   TStreamerInfo  *fInfo;          ///< Pointer to TStreamerInfo object writing/reading the buffer
+   TStreamerInfo  *fInfo{nullptr};  ///< Pointer to TStreamerInfo object writing/reading the buffer
```

Comment:
This is used in a few places in the CollectionProxy implementation.  Is it really not needed for the Text versions?

Reply:
> Is it really not needed for the Text versions?

In text version full stack of streamer infos is preserved.
One should check if **fInfo** in TBufferFile always contains that is expected.
For instance, if one store objects with other objects as data members like:

    class Test {
        int fInt;
        TNamed fNamed;
        Float16_t fFloat;
    };

Question - does **fInfo** really points on Test class after storing **fNamed** member?

Reply:
Maintaining this is the job of IncrementLevel and DecrementLevel ...

Reply:
> Maintaining this is the job of IncrementLevel and DecrementLevel ...

Now I see in TBufferFile.cxx source code. 
Actually, TBufferJSON/XML/SQL maintain own stack of streamer infos, but with additional information. Therefore I will keep logic as is